engine/source/materials/fail/tensstrain/fail_tensstrain_ib.F

```diff
@@ -0,0 +1,398 @@
+      INTEGER                     ,INTENT(IN)    :: NUPARAM ! size of parameter array
+      INTEGER                     ,INTENT(IN)    :: IPT     ! current integration point
+      INTEGER                     ,INTENT(IN)    :: IOUT    ! output file unit
+      INTEGER                     ,INTENT(IN)    :: ISTDO   ! output file unit
+      INTEGER                     ,INTENT(IN)    :: SNPC 
+      INTEGER                     ,INTENT(IN)    :: STF       
+      INTEGER                     ,INTENT(IN)    :: ISMSTR
+      INTEGER                     ,INTENT(IN)    :: NVARF   
+      INTEGER ,DIMENSION(NEL)     ,INTENT(IN)    :: NGL     ! table of element identifiers
+      INTEGER ,DIMENSION(100)     ,INTENT(IN)    :: IFUNC   ! table of functions identifiers
+      INTEGER ,DIMENSION(SNPC)    ,INTENT(IN)    :: NPF
+      INTEGER ,DIMENSION(NEL)     ,INTENT(INOUT) :: FOFF
+      my_real                     ,INTENT(IN)    :: TIME    ! current time    
+      my_real ,DIMENSION(NUPARAM) ,INTENT(IN)    :: UPARAM  ! failure model parameter array
+      my_real ,DIMENSION(NEL)     ,INTENT(IN)    :: EPSD    ! strain rate
+      my_real ,DIMENSION(NEL)     ,INTENT(IN)    :: TSTAR
+      my_real ,DIMENSION(NEL)     ,INTENT(IN)    :: AL
+      my_real ,DIMENSION(NEL)     ,INTENT(IN)    :: EPSXX 
+      my_real ,DIMENSION(STF)     ,INTENT(IN)    :: TF
+      my_real ,DIMENSION(NEL)     ,INTENT(INOUT) :: OFF     ! element desactivation flag
+      my_real ,DIMENSION(NEL)     ,INTENT(INOUT) :: DFMAX   ! maximum damage
+      my_real ,DIMENSION(NEL)     ,INTENT(INOUT) :: DAMSCL  !1-damage
+      my_real ,DIMENSION(NEL,NVARF),INTENT(INOUT) :: UVAR   
+      my_real ,DIMENSION(NEL)     ,INTENT(INOUT) :: TDEL    ! desactivation time
+C-----------------------------------------------
+C   L o c a l   V a r i a b l e s
+C-----------------------------------------------
+      INTEGER I,J,NINDXD,NINDXF,NINDX1,NINDX2,S_FLAG,SCALE_FLAG,STRDEF,STRFLAG
+      INTEGER ,DIMENSION(NEL) :: INDXD,INDXF,INDX1,INDX2
+      my_real R1,R2,EPSF1,EPSF2,LAMBDA,FAC,DF,
+     .        EL_REF,SC_EL,SCALE_TEMP,RFAC,RFAC2,
+     .        UNIT_T,EPSP_UNIT,FINTER,DYDX,EPST1,EPST2
+      EXTERNAL FINTER
+      my_real ,DIMENSION(NEL) :: EPS_MAX,DAMAGE,EPSRATE,EPS11,EPSI,EPS_EQ
+C=======================================================================
+      EPST1   = UPARAM(1)
+      EPST2   = UPARAM(2)
+      EPSF1   = UPARAM(3)            
+      EPSF2   = UPARAM(4)
+      SC_EL      = UPARAM(5)
+      EL_REF     = UPARAM(6)
+      SCALE_TEMP = UPARAM(7)
+      S_FLAG     = INT(UPARAM(8))
+      UNIT_T     = UPARAM(9)
+      STRDEF     = INT(UPARAM(10))
+c
+      DAMAGE(:NEL)  = ZERO
+      EPS_MAX(:NEL) = ZERO
+      NINDXD  = 0  
+      NINDXF  = 0  
+      STRFLAG = 0 
+C-----------------------------------------------
+c     Initialization
+C-----------------------------------------------
+      IF (UVAR(1,1) == ZERO) THEN 
+        DO I=1,NEL
+          SELECT CASE (S_FLAG)
```

Comment:
Hi @pmondalek 

I think code will be more efficient if you move the SELECT case out of the 1->NEL loop.

Can you correct this ?
Thanks,
Sébastien

Reply:
Thanks Pamela for the correction, have a nice day. Sébastien